snap/info.go

```diff
@@ -553,6 +553,33 @@ type TimerInfo struct {
 	Timer string
 }
 
+// StopModeType is the type for the "stop-mode:" of a snap app
+type StopModeType string
+
+// KillAll returns if the stop-mode means all processes should be killed
+// when the service is stopped or just the main process.
+func (st StopModeType) KillAll() bool {
+	return string(st) == "" || strings.HasSuffix(string(st), "-all")
+}
+
+// KillSignal returns the signal that should be used to kill the process
+// (or an empty string if no signal is needed)
+func (st StopModeType) KillSignal() string {
+	if st == "" {
```

Comment:
Would be safer to check if it is valid here also, as otherwise we risk passing invalid user input elsewhere we don't know about. This is probably not true right now, as the validation function is being called on the way in, but it's a belt+suspender approach. If we check here it can never go wrong.